protokt-runtime/src/main/kotlin/com/toasttab/protokt/rt/UnknownFieldSet.kt

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2020 Toast Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.toasttab.protokt.rt
+
+class UnknownFieldSet
+private constructor(
+    val unknownFields: Map<Int, Field>
```

Comment:
can this be inline so that we don't create an extra wrapper object?

Reply:
We'd have to expose the constructor. Most of the time the unknown field set will be the singleton empty object - API-wise, how do we feel about saving the extra object and exposing the constructor for this "micro-optimization" (?) vs. not exposing it?

Reply:
Per discussion - we can revisit later.